crates/guest-rust/rt/src/async_support/waitable.rs

```diff
@@ -169,17 +200,37 @@ where
     /// This relinquishes control of the original `completion_status` pointer
     /// passed to `register_waker` after this call has completed.
     pub fn unregister_waker(self: Pin<&mut Self>, waitable: u32) {
+        // SAFETY: the contract of `wasip3_task_set` is that the returned
+        // pointer is valid for the lifetime of our entire task, so it's valid
+        // for this stack frame. Additionally we assert it's non-null to
+        // double-check it's initialized and additionlly check the version for
+        // the fields that we access.
+        //
+        // Otherwise the `waitable_unregister` callback should be safe if we
+        // pass a waitable that we own and the task's own pointer.
```

Comment:
"and the tasks's own pointer." Sorry I couldn't parse that. Maybe "and *is* the task's own pointer" or something to that effect?